### ISSUE TITLE
TEST_RUN=1 ./build.bat pass, but error when ./build.bat

### ISSUE THREAD

**Issue by GerardusOrdo** [misch7/nextcloud-client-building#5]
I've been follow the instruction provided, when TEST_RUN=1 ./build.bat, it pass.
When it comes to run ./build.bat, I get some error.
My environment is :
- Windows 10 Pro 1803 64bit
- Visual Studio Community 2017 15.9.6
- Qt 5.13.1 installed with MinGW 32bit, MinGW 64bit, MSVC2015 64bit, MSVC2017 32bit, MSVC2017 64bit option checked

Error is about qt5keychain, detail is :
[error.txt](https://github.com/misch7/nextcloud-client-building/files/3706824/error.txt)
C:\Nextcloud\client-building\qtkeychain\build\Release\qt5keychain.dll : fatal error LNK1120: 64 unresolved externals [C:\Nextcloud\client-building\qtkeychain\build\qt5keychain.vcxproj]
"*** Build FAILED: qtkeychain Release Win64 (single-build-qtkeychain.bat)"
"***** Build FAILED (build.bat)"

what wrong?

**Comment by misch7** [comment]
Hey @GerardusOrdo,

sorry, I haven't seen the issue.

The active repo has moved to: https://github.com/nextcloud/client-building/

Your `error.txt` shows that there are several Qt libraries missing. My first guess would be that is has to do with some include header issues in Qt 5.13.1. Maybe you find something in their bug tracker.
This comes into my mind because I had a similar problem when I upgraded to Qt 5.12.8 this week, you can see the details in the updated `client-building` repo.

Did you manage to solve the issue?

**Comment by socomsystems** [comment]
I had to change qt5keychain.dll to qt5keychaind.dll in "single-build-qtkeychain.bat"  due to a stat issue on Qt\5.12.9.

```
echo "* Copy qt5keychaind.dll to %QT_BIN_PATH%/ for windeployqt to find it."
start "copy qt5keychaind.dll for windeployqt" /B /wait cp -f "%MY_INSTALL_PATH%/bin/qt5keychaind.dll" "%QT_BIN_PATH%/"
if %ERRORLEVEL% neq 0 goto onError
```
Which looks to cascade into a linking issue now... continuing debug.
 ```
 dummycredentials.cpp
  abstractcredentials.cpp
  credentialscommon.cpp
  Generating Code...
  Compiling...
  keychainchunk.cpp
  httpcredentials.cpp
  Generating Code...
LINK : fatal error LNK1104: cannot open file 'C:\Nextcloud\client-building\libs\Debug\qtkeychain\Win64\lib\qt5keychain.lib' [C:\Nextcloud\client-building\desktop\build\src\libsync\nextcloudsync.vcxproj]
"*** Build FAILED: desktop Debug Win64 (single-build-desktop.bat)"
"***** Build FAILED (build.bat)"
```

$ env
USERDOMAIN=TOTUS
OS=Windows_NT
COMMONPROGRAMFILES=C:\Program Files\Common Files
PROCESSOR_LEVEL=6
PSModulePath=C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\
CommonProgramW6432=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
FP_NO_HOST_CHECK=NO
LANG=en_US.UTF-8
OPENSSL_CONF=C:\Program Files\OpenSSL-Win64\bin\openssl.cfg
MSYSTEM_CARCH=x86_64
DISPLAY=needs-to-be-defined
HOSTNAME=TOTUS
PUBLIC=C:\Users\Public
OLDPWD=/c/Nextcloud
CONFIG_SITE=/mingw64/etc/config.site
EXEPATH=C:\Program Files\Git
MSYSTEM_CHOST=x86_64-w64-mingw32
USERNAME=troy
ChocolateyInstall=C:\ProgramData\chocolatey
JAVA_HOME=C:\java-se-8u41-ri
LOGONSERVER=\\TOTUS
PROCESSOR_ARCHITECTURE=AMD64
LOCALAPPDATA=C:\Users\troy\AppData\Local
COMPUTERNAME=TOTUS
SYSTEMDRIVE=C:
USERPROFILE=C:\Users\troy
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
SYSTEMROOT=C:\WINDOWS
USERDOMAIN_ROAMINGPROFILE=TOTUS
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 26 Stepping 3, GenuineIntel
MINGW_PACKAGE_PREFIX=mingw-w64-x86_64
PWD=/c/Nextcloud/client-building
SSH_ASKPASS=/mingw64/libexec/git-core/git-gui--askpass
HOME=/c/Users/troy
TMP=/tmp
TERM_PROGRAM=mintty
TERM_PROGRAM_VERSION=3.2.0
MSYSTEM_PREFIX=/mingw64
OneDrive=C:\Users\troy\OneDrive
PROCESSOR_REVISION=1a03
TMPDIR=/tmp
LTSHARE=L:
NUMBER_OF_PROCESSORS=2
ProgramW6432=C:\Program Files
COMSPEC=C:\WINDOWS\system32\cmd.exe
APPDATA=C:\Users\troy\AppData\Roaming
SHELL=/usr/bin/bash
TERM=xterm
WINDIR=C:\WINDOWS
MINGW_CHOST=x86_64-w64-mingw32
ProgramData=C:\ProgramData
SHLVL=1
PLINK_PROTOCOL=ssh
ACLOCAL_PATH=/mingw64/share/aclocal:/usr/share/aclocal
PROGRAMFILES=C:\Program Files
MANPATH=/mingw64/local/man:/mingw64/share/man:/usr/local/man:/usr/share/man:/usr/man:/share/man
ORIGINAL_TEMP=/tmp
ORIGINAL_TMP=/tmp
ALLUSERSPROFILE=C:\ProgramData
GOOGLE_DEFAULT_CLIENT_ID=no
TEMP=/tmp
DriverData=C:\Windows\System32\Drivers\DriverData
MSYSTEM=MINGW64
MINGW_PREFIX=/mingw64
asl.log=Destination=file;OnFirstLog=command,environment,parent
ANT_HOME=C:\apache-ant-1.10.7
ProgramFiles(x86)=C:\Program Files (x86)
GOOGLE_API_KEY=no
PATH=/c/Users/troy/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/mingw64/bin:/usr/bin:/c/Users/troy/bin:/c/WINDOWS/SYSTEM32:/c/WINDOWS:/c/WINDOWS/SYSTEM32/WBEM:/c/WINDOWS/SYSTEM32/WINDOWSPOWERSHELL/V1.0:/c/PROGRAM FILES (X86)/QUICKTIME/QTSYSTEM:/c/PROGRAM FILES (X86)/AMD/ATI.ACE/CORE-STATIC:/c/WINDOWS/SYSTEM32/OPENSSH:/c/PROGRAM FILES (X86)/ATI TECHNOLOGIES/ATI.ACE/CORE-STATIC:/c/apache-ant-1.10.7/BIN:/c/java-se-8u41-ri/BIN:/c/PROGRAMDATA/CHOCOLATEY/BIN:/c/PROGRAM FILES/OPENSSH-WIN64:/cmd:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/Program Files (x86)/Nmap:/c/Users/troy/AppData/Local/Microsoft/WindowsApps:/c/Program Files/OpenSSL-Win64/bin:/c/Users/troy/AppData/Local/Programs/Microsoft VS Code/bin:/usr/bin/vendor_perl:/usr/bin/core_perl
PS1=\[\033]0;$TITLEPREFIX:$PWD\007\]\n\[\033[32m\]\u@\h \[\033[35m\]$MSYSTEM \[\033[33m\]\w\[\033[36m\]`__git_ps1`\[\033[0m\]\n$
HOMEDRIVE=C:
ChocolateyLastPathUpdate=132419657330695852
PKG_CONFIG_PATH=/mingw64/lib/pkgconfig:/mingw64/share/pkgconfig
INFOPATH=/usr/local/info:/usr/share/info:/usr/info:/share/info
HOMEPATH=\Users\troy
ORIGINAL_PATH=/mingw64/bin:/usr/bin:/c/Users/troy/bin:/c/WINDOWS/SYSTEM32:/c/WINDOWS:/c/WINDOWS/SYSTEM32/WBEM:/c/WINDOWS/SYSTEM32/WINDOWSPOWERSHELL/V1.0:/c/PROGRAM FILES (X86)/QUICKTIME/QTSYSTEM:/c/PROGRAM FILES (X86)/AMD/ATI.ACE/CORE-STATIC:/c/WINDOWS/SYSTEM32/OPENSSH:/c/PROGRAM FILES (X86)/ATI TECHNOLOGIES/ATI.ACE/CORE-STATIC:/c/apache-ant-1.10.7/BIN:/c/java-se-8u41-ri/BIN:/c/PROGRAMDATA/CHOCOLATEY/BIN:/c/PROGRAM FILES/OPENSSH-WIN64:/cmd:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/Program Files (x86)/Nmap:/c/Users/troy/AppData/Local/Microsoft/WindowsApps:/c/Program Files/OpenSSL-Win64/bin:/c/Users/troy/AppData/Local/Programs/Microsoft VS Code/bin
GOOGLE_DEFAULT_CLIENT_SECRET=no
JAVA_HOME2=C:\Program Files\Java\jdk-14.0.1
_=/usr/bin/env

**Comment by misch7** [comment]
Thanks for stepping in, @socomsystems :)

This should be solved by nextcloud/client-building#39
